### PR TITLE
Using release-1.23 image for service mesh tp1

### PR DESCRIPTION
**Pull request by FilipB**
Depends on https://github.com/openshift/release/pull/56134

### CODE REVIEW

**Review by luksa — APPROVED** [review]

File: ci-operator/config/openshift-service-mesh/sail-operator/openshift-service-mesh-sail-operator-release-3.0-tp1.yaml [hunk 1]
```diff
@@ -7,7 +7,7 @@ build_root:
   image_stream_tag:
     name: maistra-builder
     namespace: ci
-    tag: upstream-1.22
```
Comment (dgn) [inline ci-operator/config/openshift-service-mesh/sail-operator/openshift-service-mesh-sail-operator-release-3.0-tp1.yaml:10]: afaict we override this in every job to be `upstream-master`. we'd need to change that to actually be using this image. but: the 1.23 image is still not available, not sure why it hasn't been mirrored yet

Comment (FilipB) [reply ci-operator/config/openshift-service-mesh/sail-operator/openshift-service-mesh-sail-operator-release-3.0-tp1.yaml:10]: Fixing thanks

**Pull request closed without merging**